src/Makefile

```diff
@@ -340,6 +344,24 @@ ifeq ($(KERNEL),Darwin)
 	LDFLAGS += -arch $(arch) -mmacosx-version-min=10.14
 endif
 
+# To cross-compile for Android, NDK version r21 or later is recommended. 
+# In earlier NDK versions, you'll need to pass -fno-addrsig if using GNU binutils.
+# Currently we don't know how to make PGO builds with the NDK yet.
+ifeq ($(COMP),ndk)
+	CXXFLAGS += -stdlib=libc++ -fPIE -DUSE_POPCNT -DUSE_NEON
+	LDFLAGS += -static-libstdc++ -fPIE -pie -lm -latomic -DUSE_POPCNT -DUSE_NEON -DNDEBUG -O3 -flto
```

Comment:
probably we shouldn't add `-O3 -flto` here (i.e. `optimize=no` builds, but at another place)

Reply:
I see. I didn't think of that. Thanks.

Reply:
For some reason, the `-O3 -flto` are not showing up when set up the same way as gcc (under the Optimization and LTO sections). 

As a workaround, I can leave them in by default, and filter-out those flags if `optimize = no` although that's not a good practice or inconsistent with the rest of the Makefile.

Reply:
in principle should work to set it in section 3.8... notice that the ifeq/else/ etc are indented somewhat confusingly in section 3.8

Reply:
So, I followed your instructions and compiled my first android app :-)
The LTO situation on the PR is actually not correct, this flag needs to be present also at compile time. 
Let me have a look if I can figure it out.

Reply:
Congrats on compiling for Android! 

 `-O3` also seems to be missing from the CXXFLAGS. I thought it would be supplied from another/existing part of the Makefile, but looks like it wasn't passed for some reason.

Reply:
I think I have this figured out, needs some reorganizing. I'll paste a diff of the changes as a comment, and maybe you can test if this still looks good to you.